clients/rospy/src/rospy/core.py

```diff
@@ -149,6 +151,57 @@ def rospywarn(msg, *args):
 logerror = logerr # alias logerr
 
 logfatal = logging.getLogger('rosout').critical
+
+
+class LoggingThrottle(object):
+
+    last_logging_time_table = {}
+
+    def __call__(self, id, logging_func, period, msg):
+        """Do logging specified message periodically.
+
+        - id (str): Id to identify the caller
+        - logging_func (function): Function to do logging.
+        - period (float): Period to do logging in second unit.
+        - msg (object): Message to do logging.
+        """
+        now = rospy.Time.now()
+
+        last_logging_time = self.last_logging_time_table.get(id)
```

Comment:
I found the way to get column number where the function is called.
http://stackoverflow.com/questions/37606094/is-there-way-to-get-function-called-column-location-not-only-file-and-line-infor/37614347#37614347